src/Proxy.NET/Routes/AzureAISearch.cs

```diff
@@ -0,0 +1,63 @@
+using System.Text.Json;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Proxy.NET.Authentication;
+using Proxy.NET.Models;
+using Proxy.NET.Services;
+
+namespace Proxy.NET.Routes;
+
+public static class AzureAISearch
+{
+    public static RouteGroupBuilder MapAzureAISearchRoutes(this RouteGroupBuilder builder)
+    {
+        builder.MapPost("/indexes/{index}/docs/search", ProcessRequestAsync);
+        builder.MapPost("/indexes('{index}')/docs/search.post.search", ProcessRequestAsync);
+        return builder;
+    }
+
+    [Authorize(AuthenticationSchemes = ProxyAuthenticationOptions.ApiKeyScheme)]
+    private static async Task<IResult> ProcessRequestAsync(
+        [FromServices] ICatalogService catalogService,
+        [FromServices] IProxyService proxyService,
+        [FromQuery(Name = "api-version")] string apiVersion,
+        [FromBody] JsonDocument requestJsonDoc,
+        HttpContext context,
+        string index
+    )
+    {
+        using (requestJsonDoc)
+        {
+            // Get the route pattern and extract the extension path that will be appended to the upstream endpoint
+            var routePattern = (context.GetEndpoint() as RouteEndpoint)?.RoutePattern.RawText;
+            var extPath = routePattern?.Split("/indexes").Last();
+            var requestContext = (RequestContext)context.Items["RequestContext"]!;
+
+            requestContext.DeploymentName = index;
+            var deployment = await catalogService.GetCatalogItemAsync(requestContext);
+
+            var url = GenerateEndpointUrl(deployment, extPath!, apiVersion);
```

Comment:
We should be null-checking the `extPath` rather than forcing it. If there's a possibility of `null`, based on how it's used, but we don't expect it to be `null`, we're needing some kind of validation somewhere.